sdk/eventhub/Azure.Messaging.EventHubs.Shared/src/Testing/TestEnvironment.cs

```diff
@@ -35,24 +35,40 @@ public static class TestEnvironment
         private static readonly Lazy<string> EventHubsSecretInstance =
             new Lazy<string>(() => ReadAndVerifyEnvironmentVariable("EVENT_HUBS_SECRET"), LazyThreadSafetyMode.PublicationOnly);
 
+        /// <summary>The environment variable value for the namespace connection string, lazily evaluated.</summary>
+        private static readonly Lazy<string> EventHubsNamespaceConnectionString =
+            new Lazy<string>(() => ReadConnectionStringFromEnvironment("EVENT_HUBS_OVERRIDE_NAMESPACE_CONNECTION_STRING"), LazyThreadSafetyMode.PublicationOnly);
```

Comment:
Thank you!   I think this makes it clear that we're going against the natural flow.